codeforamerica/vita-min
# GYR1-508 Preventing access to bank details for greeters

## [GYR1-508](https://codeforamerica.atlassian.net/browse/GYR1-508)
## Is PM acceptance required? (delete one)
- Yes - don't merge until JIRA issue is accepted!
## What was done?
- Added a check to return a 404 if a greeter accesses the BankAccountsController
- Added a check in the clients show.html.erb to hide the bank details section if the user is a greeter.
- Added specs to cover cases where a user can view bank details and when they can't.
## How to test?
- Test on heroku.
- Create a client that has bank details.
- Try to view the client as a non greeter, and you should see the bank details.
- Try to view the client as a greeter and you should not see the bank details
## Risk Assessment
  - Low risk - preventing access to something that was accessible before. Specs mitigate inadvertent access removal

### Greeter does not see bank details
<img width="1095" alt="image" src="https://github.com/user-attachments/assets/52c90cf7-2473-47c3-9554-7846a5c9c762">

### Others still see bank details
<img width="783" alt="image" src="https://github.com/user-attachments/assets/d3e5eaff-3648-4a0d-87cc-63a4eacabaa3">




[GYR1-508]: https://codeforamerica.atlassian.net/browse/GYR1-508?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
